homeassistant/components/smtp/notify.py

```diff
@@ -223,14 +223,31 @@ def _send_email(self, msg, recipients):
             try:
                 mail.sendmail(self._sender, recipients, msg.as_string())
                 break
-            except smtplib.SMTPServerDisconnected:
-                _LOGGER.warning(
-                    "SMTPServerDisconnected sending mail: retrying connection"
+            except (
+                smtplib.SMTPSenderRefused,
+                smtplib.SMTPDataError,
+                smtplib.SMTPConnectError,
+                smtplib.SMTPHeloError,
+                smtplib.SMTPResponseException,
+                smtplib.SMTPServerDisconnected,
+                smtplib.SMTPRecipientsRefused,
+                smtplib.SMTPNotSupportedError,
+                smtplib.SMTPException,
```

Comment:
`SMTPException` is the base exception class, so you don't need to enumerate all of these exceptions, you can just have the generic one, and `__name__` should still return the correct class name

Reply:
You're right and I had indeed considered that, but I decided the extra few lines were worth the added readability and maintainability. 

Enumerating them all could make it easier for others in the future to understand which exceptions were considered when I wrote the code, so adding additional functionality for one of the specific exceptions might be easier, if for example they could instantly know the code does not handle some other exception.

Only catching the base SMTPException could make the code less clear. This was effectively the problem I was remedying in the first place, since the original code just showed it catching a single exception and it took extra time to figure out how it related to the other ones.

Reply:
@raman325 is probably right here to catch only the base exception as if other errors are added down the line it's still going to get caught as the base exception is in the list.
Don't think the list of exceptions provides any necessary value as you want to capture all of them anyhow to provide to logger output.

Reply:
agreed with @gjohansson-ST - it's clear from the code that you've made the exception logging generic for all types. If you are concerned about readability, you can also add a comment that just says this is the base exception class and that details will be logged, but it took me all of 30 seconds to understand that was your intention.